### DISCUSSION TITLE
在macos运行是报 Common Language Runtime detected an invalid program.

### DISCUSSION THREAD

**Discussion by njroyin** [dotnetcore/BootstrapBlazor#4429] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

在macos使用BootstrapBlazor8.10.1，运行到builder.Services.AddBootstrapBlazor()时报错Common Language Runtime detected an invalid program.降级到8.9.4可正常运行。
<img width="840" alt="image" src="https://github.com/user-attachments/assets/35fa049a-b05e-4fa7-a0e5-155a8f4ea472">

### Expected Behavior

处理报错

### Interactive render mode

Interactive Auto (Server, then client)

### Steps To Reproduce

[AppGlobal.zip](https://github.com/user-attachments/files/17336733/AppGlobal.zip)

### .NET Version

8.0.402

**Comment by bb-auto** [comment]
@njroyin Thank you for reporting. We will give triage later.

**Comment by ArgoZhang** [comment]
@njroyin 很抱歉，根据提供的信息与复现工程。我这里无法复现

未找到 `8.0.402` 版本的 SDK 我使用的是 `8.0.403` 版本进行的测试

![image](https://github.com/user-attachments/assets/0be02ea1-0131-4d82-bc01-fa809cdad7e1)

**Reply by njroyin** [reply]
我在使用net sdk 8.0.403也会出现该错误，不过将BootstrapBlazor升级到8.10.2就没有出现该错误.

**Reply by ArgoZhang** [reply]
@njroyin 感觉是环境的问题，你一直没有提供 `dotnet --info` 感觉你的问题是在开发环境中出现的吧

**Comment by njroyin** [comment]
.NET SDK:
 Version:           8.0.403
 Commit:            c64aa40a71
 Workload version:  8.0.402.1
 MSBuild version:   17.11.9+a69bbaaf5

运行时环境:
 OS Name:     Mac OS X
 OS Version:  15.0
 OS Platform: Darwin
 RID:         osx-arm64
 Base Path:   /usr/local/share/dotnet/sdk/8.0.403/

已安装 .NET 工作负载:
配置为在安装新清单时使用 workload sets。
 [wasm-tools]
   安装源文件: SDK 8.0.400
   清单版本:    8.0.8/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.workload.mono.toolchain.current/8.0.8/WorkloadManifest.json
   安装类型:        FileBased

 [macos]
   安装源文件: SDK 8.0.400
   清单版本:    15.0.8303/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.macos/15.0.8303/WorkloadManifest.json
   安装类型:        FileBased

 [ios]
   安装源文件: SDK 8.0.400
   清单版本:    18.0.8303/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.ios/18.0.8303/WorkloadManifest.json
   安装类型:        FileBased

 [maccatalyst]
   安装源文件: SDK 8.0.400
   清单版本:    18.0.8303/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.maccatalyst/18.0.8303/WorkloadManifest.json
   安装类型:        FileBased

 [maui]
   安装源文件: SDK 8.0.400
   清单版本:    8.0.82/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.maui/8.0.82/WorkloadManifest.json
   安装类型:        FileBased

 [tvos]
   安装源文件: SDK 8.0.400
   清单版本:    18.0.8303/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.tvos/18.0.8303/WorkloadManifest.json
   安装类型:        FileBased

 [android]
   安装源文件: SDK 8.0.400
   清单版本:    34.0.138/8.0.100
   清单路径:       /usr/local/share/dotnet/sdk-manifests/8.0.100/microsoft.net.sdk.android/34.0.138/WorkloadManifest.json
   安装类型:        FileBased


Host:
  Version:      9.0.0-rc.1.24431.7
  Architecture: arm64
  Commit:       static

.NET SDKs installed:
  6.0.423 [/usr/local/share/dotnet/sdk]
  7.0.410 [/usr/local/share/dotnet/sdk]
  8.0.402 [/usr/local/share/dotnet/sdk]
  8.0.403 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.31 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.8 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 9.0.0-rc.1.24452.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.31 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.8 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 9.0.0-rc.1.24431.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  None

Environment variables:
  Not set

global.json file:
  /Users/luoyin/Documents/RiderProjects/AppGlobal/global.json

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

**Reply by ArgoZhang** [reply]
看起来环境没问题

**Reply by njroyin** [reply, marked as answer]
嗯，不管了，我将BootstrapBlazor升级到8.10.2就可以了。多谢
